src/main/kotlin/app/batch/Validator.kt

```diff
@@ -0,0 +1,106 @@
+package app.batch
+
+import app.batch.processor.DecryptionProcessor
+import app.domain.DecryptedRecord
+import app.domain.SourceRecord
+import app.exceptions.BadDecryptedDataException
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import java.text.SimpleDateFormat
+import java.util.*
+
+@Component
+class Validator {
+
+    fun skipBadDecryptedRecords(item: SourceRecord, decrypted: String): DecryptedRecord? {
+        val hbaseRowKey = Arrays.copyOfRange(item.hbaseRowId, 4, item.hbaseRowId.size)
+        val hbaseRowId = String(hbaseRowKey)
+        val db = item.db
+        val collection = item.collection
+        try {
+            val jsonObject = parseDecrypted(hbaseRowId, decrypted)
+            if (null != jsonObject) {
+                retrieveId(hbaseRowId, jsonObject)
+                val lastUpdatedTimestamp = retrievelastUpdatedTimestamp(hbaseRowId, jsonObject)
+                lastUpdatedTimestamp?.let { validate(hbaseRowId, lastUpdatedTimestamp) }
+                jsonObject.addProperty("timestamp", item.hbaseTimestamp)
+                return DecryptedRecord(jsonObject, db, collection)
+            }
+        } catch (e: Exception) {
+            val ex = BadDecryptedDataException(hbaseRowId, db, collection, e.message!!)
+            DecryptionProcessor.logger.error(ex.message)
```

Comment:
Can it have it's own logger so that the messages in the log indicate the correct class